prune --dry-run on ARM fails with segmentation violation

Hardware:
--------------------------
"cubietruck 2" with 1GB RAM


Output of `restic version`
--------------------------
restic 0.11.0-dev (compiled manually) compiled with go1.15.6 on linux/arm


How did you run restic exactly?
--------------------------

```
# sudo restic_v0.11.0-207-gf7c7c2f7_linux_arm -r repo prune --dry-run
enter password for repository: 
repository 551da140 opened successfully, password is correct
loading indexes...
loading all snapshots...
finding data that is still in use for 11 snapshots
panic: runtime error: invalid memory address or nil pointer dereference
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11ce8]

goroutine 14 [running]:
runtime/internal/atomic.goLoad64(0x1824294, 0x0, 0x1a6e200)
        /usr/local/go/src/runtime/internal/atomic/atomic_arm.go:131 +0x1c
github.com/restic/restic/internal/ui/progress.(*Counter).get(...)
        /home/beta/restic.git/internal/ui/progress/counter.go:69
github.com/restic/restic/internal/ui/progress.(*Counter).run.func1(0x1824270)
        /home/beta/restic.git/internal/ui/progress/counter.go:75 +0x24
panic(0x88b8b8, 0xf0e838)
        /usr/local/go/src/runtime/panic.go:969 +0x158
runtime/internal/atomic.goLoad64(0x1824294, 0x1af773c, 0x3)
        /usr/local/go/src/runtime/internal/atomic/atomic_arm.go:131 +0x1c
github.com/restic/restic/internal/ui/progress.(*Counter).get(...)
        /home/beta/restic.git/internal/ui/progress/counter.go:69
github.com/restic/restic/internal/ui/progress.(*Counter).run(0x1824270)
        /home/beta/restic.git/internal/ui/progress/counter.go:90 +0x6c
created by github.com/restic/restic/internal/ui/progress.New
        /home/beta/restic.git/internal/ui/progress/counter.go:46 +0x10c
```

What backend/server/service did you use to store the repository?
--------------------------
I used restic rest-server to store the repo

--- a/internal/ui/progress/counter.go
+++ b/internal/ui/progress/counter.go
@@ -3,7 +3,6 @@ package progress
 import (
 	"os"
 	"sync"
-	"sync/atomic"
 	"time"
 
 	"github.com/restic/restic/internal/debug"
@@ -25,7 +24,9 @@ type Counter struct {
 	stopped chan struct{} // Closed by run.
 	stop    chan struct{} // Close to stop run.
 	tick    *time.Ticker
-	value   uint64
+
+	valueMutex sync.Mutex
+	value      uint64
 }
 
 // New starts a new Counter.
@@ -52,7 +53,10 @@ func (c *Counter) Add(v uint64) {
 	if c == nil {
 		return
 	}
-	atomic.AddUint64(&c.value, v)
+
+	c.valueMutex.Lock()
+	c.value += v
+	c.valueMutex.Unlock()
 }
 
 // Done tells a Counter to stop and waits for it to report its final value.
@@ -66,7 +70,13 @@ func (c *Counter) Done() {
 	*c = Counter{} // Prevent reuse.
 }
 
-func (c *Counter) get() uint64 { return atomic.LoadUint64(&c.value) }
+func (c *Counter) get() uint64 {
+	c.valueMutex.Lock()
+	v := c.value
+	c.valueMutex.Unlock()
+
+	return v
+}
 
 func (c *Counter) run() {
 	defer close(c.stopped)

